src/Pass/IframeBrightCoveTagTransformPass.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/*
+ * Copyright 2016 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+namespace Lullabot\AMP\Pass;
+
+use QueryPath\DOMQuery;
+
+use Lullabot\AMP\Utility\ActionTakenLine;
+use Lullabot\AMP\Utility\ActionTakenType;
+
+/**
+ * Class IframeBrighCoveTagTransformPass
+ * @package Lullabot\AMP\Pass
+ *
+ * Transform all BrighCove embed code <iframe> tags which don't have noscript as an ancestor to <amp-brighcove> tags
+ *
+ * This is what a BrighCove embed looks like:
+ *   <iframe width="560" height="315" src="https://www.BrighCove.com/embed/MnR9AVs6Q_c" frameborder="0" allowfullscreen></iframe>
+ *
+ * @see https://github.com/ampproject/amphtml/blob/master/extensions/amp-BrighCove/amp-BrighCove.md
+ * @see https://developers.google.com/BrighCove/iframe_api_reference
+ *
+ */
+class IframeBrighCoveTagTransformPass extends BasePass
+{
+    /**
+     * A standard BrighCove video aspect ratio
+     * @var float
+     */
+    const DEFAULT_ASPECT_RATIO = 1.7778;
+    const DEFAULT_VIDEO_WIDTH = 560;
+    const DEFAULT_VIDEO_HEIGHT = 315;
+
+    function pass()
+    {
+        $all_iframes = $this->q->find('iframe:not(noscript iframe)');
+        /** @var DOMQuery $el */
+        foreach ($all_iframes as $el) {
+            /** @var \DOMElement $dom_el */
+            $dom_el = $el->get(0);
+            if (!$this->isBrighCoveIframe($el)) {
+                continue;
+            }
+
+            $lineno = $this->getLineNo($dom_el);
+            $context_string = $this->getContextString($dom_el);
+            $brightcove_code = $this->getBrightcoveCode($el);
+
+            // If we couldnt find a BrighCove videoid then we abort
+            if (empty($brightcove_code)) {
+                continue;
+            }
+
+            if ($el->hasAttr('class')) {
+                $class_attr = $el->attr('class');
+            }
+
+            /** @var \DOMElement $new_dom_el */
+            $el->after("<amp-brightcove data-videoid=\"$brightcove_code\" layout=\"responsive\"></amp-brightcove>");
```

Comment:
This DOM element is invalid.

* `data-videoid` needs to be `data-video-id`
* Mandatory attribute `data-account` is missing

See https://www.ampproject.org/docs/reference/components/amp-brightcove